Assets/SEETests/TestSourceRangeIndex.cs

```diff
@@ -0,0 +1,224 @@
+            AssertNotFound(index, c1m1M3M1, 57);
+
+            AssertNotFound(index, c1m2, 59);
+            AssertFound(index, c1m2, 60);
+            AssertFound(index, c1m2, 61);
+            AssertFound(index, c1m2, 62);
+            AssertFound(index, c1m2, 63);
+            AssertFound(index, c1m2, 64);
+            AssertNotFound(index, c1m2, 65);
+
+            AssertNotFound(index, c1m5, 65);
+            AssertFound(index, c1m5, 66);
+            AssertNotFound(index, c1m5, 67);
+        }
+
+        [Test]
+        public void TestConsistent2()
+        {
+            // This node should fit.
+            Node c1m1M4 = Child(g, c1m1, "c1.m1.M4", type: "Method", directory: "mydir/", filename: "myfile.java", line: 54, length: 1);
+
+            SourceRangeIndex index = new(g);
+            Assert.IsTrue(index.IsIsomorphic());
+            AssertFound(index, c1m1M4, 54);
+        }
+
+        /// <summary>
+        /// Asserts that either no node can be found in the file containing <paramref name="unexpected"/>
+        /// at the given <paramref name="line"/> or -- if one is found -- that the found node is
+        /// different from <paramref name="unexpected"/>.
+        /// </summary>
+        /// <param name="index">where to search</param>
+        /// <param name="unexpected">node not expected to be found</param>
+        /// <param name="line">source line where to search</param>
+        private static void AssertNotFound(SourceRangeIndex index, Node unexpected, int line)
+        {
+            Assert.IsTrue(!index.TryGetValue(unexpected.Path(), line, out Node node) // no node found at all
+                || (node != null && unexpected != node)); // if one is found, it must be different from unexpected
+        }
+
+        private static void AssertFound(SourceRangeIndex index, Node expected, int line)
+        {
+            Assert.IsTrue(index.TryGetValue(expected.Path(), line, out Node found));
+            Assert.AreEqual(expected, found);
+        }
+
+        [Test]
+        public void TestNonHomormorphicGraph()
+        {
+            // This node is logically in c1m1, but spatially nested in c1m1M3
+            Child(g, c1m1, "c1.m1.M4", type: "Method", directory: "mydir/", filename: "myfile.java", line: 56, length: 1);
+
+            SourceRangeIndex index = new(g);
+            Assert.IsFalse(index.IsIsomorphic());
+
+            LogAssert.Expect(LogType.Error, new Regex(@"Range c1.m1.M4.* is subsumed by c1.m1.M3.M1.*"));
+        }
+
+        [Test]
+        public void TestSubsumption()
+        {
+            // This node is subsumed by c1.m1.M2, but is not in parent-child relation in the node hierarchy
+            Child(g, c1m1, "c1.m1.M4", type: "Method", directory: "mydir/", filename: "myfile.java", line: 53, length: 1);
+
+            SourceRangeIndex index = new(g);
+            Assert.IsFalse(index.IsIsomorphic());
+
+            LogAssert.Expect(LogType.Error, new Regex(@"Range .* is subsumed by .*"));
```

Comment:
I don't think this needs to be a verbatim string:
```suggestion
            LogAssert.Expect(LogType.Error, new Regex("Range .* is subsumed by .*"));
```